Student/StudentE2ETests/Offline/FeatureFlagOfflineTests.swift

```diff
@@ -0,0 +1,114 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import TestsFoundation
+
+class FeatureFlagOfflineTests: OfflineE2ETest {
+    override func setUp() {
+        // Enable discussion redesign feature flag before app start
+        let featureFlagResponse = seeder.setFeatureFlag(featureFlag: .newDiscussion, state: .allowedOn)
+        XCTAssertEqual(featureFlagResponse.state, DSFeatureFlagState.allowedOn.rawValue)
+
+        super.setUp()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        // Disable discussion redesign feature flag
+        let featureFlagResponse = seeder.setFeatureFlag(featureFlag: .newDiscussion, state: .off)
+        XCTAssertEqual(featureFlagResponse.state, DSFeatureFlagState.off.rawValue)
```

Comment:
The `super` call order order should be inverted.
```
super.setup()
subclass.setup()
subclass.tearDown()
super.tearDown()
```

Reply:
This is intentional here. This is an offline test, which means that the network connection gets switched back in the superclass teardown call, while here in this teardown call a request gets sent (which is not possible if the network connection is still off) so this is why I called the teardown of the superclass first.

Reply:
Done anyway. 🙂